models/sudoswap/ethereum/sudoswap_ethereum_pools.sql

```diff
@@ -0,0 +1,275 @@
+      cardinality(_initialNFTIDs) AS initial_nft_count,
+      tx.value / 1e18 AS initial_eth
+    FROM
+      {{ source('sudo_amm_ethereum','LSSVMPairFactory_call_createPairETH') }} cre
+      INNER JOIN {{ source('ethereum','transactions') }} tx ON tx.hash = cre.call_tx_hash
+    WHERE
+      call_success
+    {% if is_incremental() %}
+    -- this filter will only be applied on an incremental run. We only want to update with new transactions.
+    AND tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+  ),
+  most_recent_spot_delta AS (
+    SELECT
+      COALESCE(del.contract_address, spot.contract_address) AS pair_address,
+      del.delta,
+      spot.spot_price
+    FROM
+      (
+        SELECT
+          *
+        FROM
+          (
+            SELECT
+              contract_address,
+              newDelta / 1e18 AS delta,
+              row_number() over (
+                PARTITION BY
+                  contract_address
+                ORDER BY
+                  evt_block_number DESC,
+                  evt_index DESC
+              ) as most_recent
+            FROM
+              {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_DeltaUpdate') }}
+          ) a
+        WHERE
+          most_recent = 1
+      ) del
+      FULL OUTER JOIN (
+        SELECT
+          *
+        FROM
+          (
+            SELECT
+              contract_address,
+              newSpotPrice / 1e18 AS spot_price,
+              row_number() over (
+                PARTITION BY
+                  contract_address
+                ORDER BY
+                  evt_block_number DESC,
+                  evt_index DESC
+              ) AS most_recent
+            FROM
+              {{ source('sudo_amm_ethereum','LSSVMPair_general_evt_SpotPriceUpdate') }}
+          ) a
+        WHERE
+          most_recent = 1
+      ) spot ON del.contract_address = spot.contract_address
+  ),
+  erc721_balances AS (
+    SELECT
+      pair_address AS holder_address,
+      SUM(
+        CASE
+          WHEN et.to = p.pair_address THEN 1
+          ELSE -1
+        END
+      ) AS tokens_held
+    FROM
+      {{ source('erc721_ethereum','evt_transfer') }} et
+      INNER JOIN pairs_created p ON p.nft_contract_address = et.contract_address
+      AND (
+        et.to = p.pair_address
+        OR et.from = p.pair_address
+      )
+    {% if is_incremental() %}
+    -- this filter will only be applied on an incremental run. We only want to update with new transfers.
+    WHERE et.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+    GROUP BY
+      1
+  ),
+  eth_balances AS (
+    WITH
+      eth_in AS (
+        SELECT
+          tr.to AS holder_address,
+          SUM(tr.value / 1e18) AS eth_funded
+        FROM
+          {{ source('ethereum','traces') }} tr
+          JOIN pairs_created pc ON pc.pair_address = tr.to
+        WHERE
+          tr.block_time > '2022-04-23'
```

Comment:
rather than having this filter every time, we've approached project start date filters in "not incremental" logic:
          {% if not is_incremental() %}
          AND tr.block_time > '2022-04-23'
          {% endif %}

Reply:
this "not incremental" logic can then also be used on all instances where "is incremental" is used, to help expedite initial build and any potential full refreshes in the future

Reply:
base tables, decoded tables, etc

Reply:
one more thing: you can also apply '2022-04-23' as a variable at the top of the model, then reference in these joins. an example:
https://github.com/duneanalytics/spellbook/pull/1472/files#diff-8be4351b64f13bdb3ced1d5365aa0de0fb03ecafcbeb1540561d00998748a1cbR12

Reply:
ok ive made the start date a variable and added is not incremental logic